Make the event package domain-agnostic

jraddaoui opened the pull request:
Move serializers and publish functions to the respective ingest and storage packages, remove helpers and any other domain specific code from the event package and update tests. Update configuration for a better separation between domains.

Refs #1222.

sevein commented on internal/ingest/event.go:
```diff
@@ -0,0 +1,59 @@
+}
+
+// PublishEvent publishes an ingest event with type safety.
+func PublishEvent(ctx context.Context, svc event.Service[*goaingest.IngestEvent], event any) {
```
I wouldn't say that it's type-safe because of the runtime type check that could lead to a panic. Here's an example using a compile-time contraint that you may want to consider: https://gist.github.com/sevein/0cd28e149ab9f4c1317531d9c3da0979.

I've never understood why Goa doesn't export an interface that can be used for cases like this. Maybe there's something that could be contributed to Goa in that regards? Perhaps at another time.

jraddaoui replied:
Thanks @sevein, that looks parfait :ok_hand:!

sevein reviewed (approved):
Thanks for doing this, it looks great! Just one minor suggestion.

jraddaoui committed:
Make the event package domain-agnostic

Move serializers and publish functions to the respective ingest and
storage packages, remove helpers and any other domain specific code
from the event package and update tests. Update configuration for a
better separation between domains.
